### PR TITLE
Ensure returned versionId is a string when listing objects

### PR DESCRIPTION
**Pull request by fflorent**
Returned versionId is a string when calling statObject(), as it is read without transformation (casting) from the headers.

However, the xml parser casts automatically what looks like a number in the responses. Consequentially, when the S3 storage provider returns a VersionId with only numbers, listObjects() returns an object with a versionId as a number, which is inconsistent with what returns other methods like statObjects() and may result in erratic behaviors for the program using these methods.

### CODE REVIEW

**Comment by trim21** [comment]
maybe https://github.com/minio/minio-js/pull/1133 ?

**Comment by fflorent** [comment]
> maybe #1133 ?

I don't think so, in my case versionIds are in base 10:
```js
> const { XMLParser } = require("fast-xml-parser")
undefined
> const  fxp = new XMLParser({
...   numberParseOptions: {
...     leadingZeros: false,
...     hex: false,
...   }
... })
undefined
> fxp.parse('<Key>0x2f</Key>') // results in string as expected
{ Key: '0x2f' }
> fxp.parse('<Key>1234</Key>') // This number encoded in base 10 results in number, I would like to force them to be string for versionId
{ Key: 1234 }
```

**Comment by trim21** [comment]
then I think we should create a new parser and disable all number, not convert it back to string...

**Comment by trim21** [comment]
> Should we then convert numbers manually?

Yes, what I mean

**Comment by trim21** [comment]
fox example, `versionId` like hex or 0123, `versionId.toString()` it not expected

**Comment by fflorent** [comment]
Right, however, the only way I found to discard automatic conversion is to use the `skipLike` option, which is hackish:
```ts
  const parseXml = new XMLParser({
    numberParseOptions: {
      skipLike: /./
    }
  })
```

Documentation of numberParseOption: https://github.com/NaturalIntelligence/fast-xml-parser/blob/edb30ddefb9c131a724761b5f79cd4f883736214/docs/v4/2.XMLparseOptions.md#numberparseoptions

Related issue: https://github.com/NaturalIntelligence/fast-xml-parser/issues/516

Is it fine for you?

**Comment by trim21** [comment]
yes, looks fine

**Comment by fflorent** [comment]
Done!

I did not make the same change for parseListObjectsV2 and parseListObjectsV2WithMetadata API as they don't have VersionId attribute to handle and keys / prefixes are parsed as string.

Do you think it's worth to make the same change as well despite of this?

File: src/xml-parsers.js [hunk 1]
```diff
@@ -236,7 +246,12 @@ export function parseListObjects(xml) {
   }
   let isTruncated = false
   let nextMarker, nextVersionKeyMarker
-  const xmlobj = parseXml(xml)
+  const fxp = new XMLParser({
```
Comment (trim21) [inline src/xml-parsers.js:249]: maybe move this outside function?

Comment (fflorent) [reply src/xml-parsers.js]: Done

Comment (trim21) [reply src/xml-parsers.js]: parser can be re-used, not very necessary to create new one for each parsing

**Comment by trim21** [comment]
@prakashsvmx

File: src/xml-parsers.js [hunk 2]
```diff
@@ -201,12 +216,13 @@ const formatObjInfo = (content, opts = {}) => {
...
   const lastModified = new Date(toArray(LastModified)[0])
   const etag = sanitizeETag(toArray(ETag)[0])
+  const size = sanitizeSize(Size)
 
   return {
     name,
     lastModified,
     etag,
-    size: Size,
+    size,
```
Comment (prakashsvmx) [inline src/xml-parsers.js:225]: here only size is updated. but the PR title is for versionId. am i missing something ?
Could you clarify @fflorent  ?

Comment (trim21) [reply src/xml-parsers.js:225]: This PR disable number parsing of xml parser, so the real number (size) need to be parsed manually.

**Review by prakashsvmx — APPROVED** [review]

**Review by kaankabalak — APPROVED** [review]

**Pull request merged**